tiledb/sm/query/readers/reader_base.h

```diff
@@ -173,7 +173,7 @@ class ReaderBase : public StrategyBase {
   /* ********************************* */
 
   /** The query condition. */
-  QueryCondition& condition_;
+  std::optional<QueryCondition> condition_;
```

Comment:
Can we just store a reference to the condition in the query (std::optional<QueryCondition>& condition_;). That way we don't create a copy? The lifetime of the query is guaranteed to be greater than the strategy.

Reply:
Good catch. I'll give it a whirl but I don't see why it wouldn't work.

Reply:
Done and done.